frontend/src/pages/modelServing/screens/global/InferenceServiceTable.tsx

```diff
@@ -1,28 +1,30 @@
 import * as React from 'react';
-import { TableComposable, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
-import { Button } from '@patternfly/react-core';
-import { GetColumnSort } from '~/utilities/useTableColumnSort';
-import { InferenceServiceKind, ServingRuntimeKind } from '~/k8sTypes';
+import { Button, ToolbarItem } from '@patternfly/react-core';
 import ManageInferenceServiceModal from '~/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal';
 import { ModelServingContext } from '~/pages/modelServing/ModelServingContext';
-import { getGlobalInferenceServiceColumns, getProjectInferenceServiceColumns } from './data';
+import Table from '~/components/Table';
+
+import { InferenceServiceKind, ServingRuntimeKind } from '~/k8sTypes';
 import InferenceServiceTableRow from './InferenceServiceTableRow';
+import { getGlobalInferenceServiceColumns, getProjectInferenceServiceColumns } from './data';
 import DeleteInferenceServiceModal from './DeleteInferenceServiceModal';
 
 type InferenceServiceTableProps = {
   clearFilters?: () => void;
   inferenceServices: InferenceServiceKind[];
   servingRuntimes: ServingRuntimeKind[];
-  getColumnSort: GetColumnSort;
   refresh: () => void;
+  enablePagination?: boolean;
+  toolbarContent?: React.ReactElement<typeof ToolbarItem>;
```

Comment:
Interesting, any reason you didn't take my "infer type from Table prop"?

Curious if it looks bulky or if you had a specific objection to it.

Reply:
do you mean like this or merging the types
```tsx
  enablePagination?: TableProps<InferenceServiceKind>["enablePagination"];
  toolbarContent?:TableProps<InferenceServiceKind>["toolbarContent"]
ts

Reply:
Content props (that take JSX) are often better to infer... same goes for functions that have complex signatures. Booleans are up to you -- but if you're threading props through, it doesn't hurt to just infer. Saves the extra step if we ever change the prop later.

You could also use `Pick` (antithesis of `Omit`) and just say "these are being threaded" https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys

You may want to wrap it in `Partial` if it doesn't do it for you 🤔 https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype

Reply:
Ok, I used pick, as it seemed the most "future proof"

Reply:
I created a playground to test -- see it [here](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcA5FDvmQNwCwAUIzAJ5hZwCyLACsWOgC8cAN6M4EuKhJYAKgAtgAOwDmALjgAjCBAA2OJfQaSpMgMKLdAEzMQlMLPY3Y8MAHQv8AOQhWsRgF9GRgB6ELgAdXlkeGACOABJOAB3ZHsUKytgGGA7ZF04MH50GGj4BLJSACsAV1R4UspkLNU4RogalXkAfkZcO3q4WR1dTWQoBIdSYQAKAEo4QQA+UThKGBqoJTglGt1dGjgAw7DTEA5UdlxgfLa2Dgh4-vA7R3hkjlT7YIZWdjhbC8lG8EgBVVBYPgQASLUTiSTSc6lZQqBJWOwwLxYLBWDS7ECaLBQIwmdEKFEabR6AwkyTQnJ5XS2exvbrOKjuTwwACi+nO9gAPH8sI8hiMxhMpktAj9hVwWABBfA1fJQmHCMTGBEycmqSkjGnwiSIrAWYDWZkOJyYDkeDk+PyBOAAMjgPGAuAA1gLAZBgfYwRC1agADTkMmKVRkOAAH3I9NySnylreZGlP36SkGIEVytV-HByBUWA03CVG3z0KEcK1xp1kfUbSgNSwIaNZ1NlhsGLeeL2ujbtbgEYpTZbYdOSQAIgB5HbYqzDiC6lRwAC0a7g2RSaDWWAAjjVgJQrO3J0ulGR4MCcXAE4yU+kN1v3rv70ndIxjj9TlEYlv4iSL4GnkLBSBgCAtA4fICnfW4CGgOAczgOosFwNAsF6JhfnueVyxVXQADF+xYGcwAZD9uQADxgKBkGDWFNRME0V31ak0lpOtzjNC0e2tLk7VcB1-C-F03XGHJ8gFd0vR9Eg-RBcFIWKMMyBHKNY3jcjE2TPiYDTdNsMzbNcwrXQ1ULYsACZS1MgjiP2UjtMZajaPo4pGPbFiGw0Wjxy88wu0fGA+32QcTFOdTGz81s4HPdFL2vBclxXddN23VJ0Dgz8h3iuwr3nW9suCtKXx3LLnI-L8aCAA).